crates/cranelift/src/lib.rs

```diff
@@ -235,13 +235,14 @@ fn get_function_address_map<'data>(
     let instructions = if let Some(ref mcr) = &context.mach_compile_result {
         // New-style backend: we have a `MachCompileResult` that will give us `MachSrcLoc` mapping
         // tuples.
-        collect_address_maps(
-            body_len,
-            mcr.buffer
-                .get_srclocs_sorted()
-                .into_iter()
-                .map(|&MachSrcLoc { start, end, loc }| (loc, start, (end - start))),
-        )
+        let mut loc_start_len_tuples: Vec<_> = mcr
+            .buffer
+            .get_srclocs_sorted()
```

Comment:
Out of curiosity, is there a speedier accesssor for non-sorted? If we do our own sort afterwards the sort here shouldn't matter too much.

Reply:
Fixed instead by actually sorting by the right thing in `get_srclocs_sorted()`!